### ISSUE TITLE
Fix UCX support for cuML/RAFT in wheels packages

### ISSUE THREAD

**Issue by pentschev** [rapidsai/build-planning#57]
In the context of adding support for UCXX in RAFT and cuML, we've found out that UCX wheels support in those libraries is broken. So far there are no UCX tests running in CI and probably few people running wheels also run UCX, and thus this problem has gone unnoticed possibly for months or even since the existence of wheels packages.

I've attempted to enable UCX-Py tests in https://github.com/rapidsai/cuml/pull/5843 but they segfault, the PR is a simplification of https://github.com/rapidsai/cuml/pull/5697 where both UCXX and UCX-Py are being added, but both segfault similarly and also depends on https://github.com/rapidsai/raft/pull/1983, but before merging the RAFT PR we want to make sure that cuML runs appropriately with UCXX and thus requires UCX support to be fixed.

I've earlier this week discussed with @vyasr about this issue and ideally we would have it fixed by 24.06, since it blocks UCXX from being tested across RAPIDS and thus delays it schedule once more. One way to resolve this would be making UCX a proper wheels package which would then be dynamically linked to, a repo for that has been created in https://github.com/rapidsai/ucx-wheels, and the latest news I had was that @msarahan would be working on that sometime soon.

@mmccarty as per your request. 🙂

**Comment by vyasr** [comment]
I'll take a look at the initial steps here since the UCX wheel build will probably be different enough from the other RAPIDS packages that the guidance I've given Michael so far may not apply, and since I have a pretty good idea of what it'll take to get it off the ground so that this particular wheel doesn't slip given the relative urgency for Peter's work. I'll try and get something minimal working ASAP so that we have confidence that we'll be able to move forward (otherwise we'll probably want to find another way to unblock Peter's work for 24.06, but I'd really like to avoid that since it's guaranteeing duplicate work). I'll coordinate a handoff with Michael soon too, though; he's getting far enough along with the corresponding libcudf work that a handoff shouldn't be hard.

**Comment by vyasr** [comment]
I have UCX wheels building in https://github.com/rapidsai/ucx-wheels/pull/1. They needed a couple of tricky tweaks, but they nominally seem to work now. I'm testing them out with ucxx in https://github.com/rapidsai/ucxx/pull/226.

**Comment by vyasr** [comment]
UCX wheels are now available, and the ucxx PR has been updated to use those wheels directly from the nightly index. CI is passing on the ucxx PR. We should be able to get that in soon. I've requested that someone from the build-infra team take a look at porting similar changes to ucx-py. If that is also urgent for 24.06, then it would probably be good to collaborate with Peter on that work to get it done as fast as possible.

**Comment by jameslamb** [comment]
> I've requested that someone from the build-infra team take a look at porting similar changes to ucx-py

@vyasr and I talked about this and I'm gonna do this one.

**Comment by jameslamb** [comment]
Correct me if I'm wrong @pentschev , but I *think* all that remains for this issue is https://github.com/rapidsai/cuml/pull/5697, right?

**Comment by vyasr** [comment]
Presumably we want something similar for cugraph, but otherwise yes I think that's right.

**Comment by sisodia1701** [comment]
@pentschev given the testing portion of this issue remains which I believe is being worked by you, I am adding you as a co-assignee on this issue apart from James.

**Comment by vyasr** [comment]
@pentschev do you plan to put together a cugraph PR for testing ucx-py/ucxx? Once that's done I believe that this issue can be closed.

**Comment by pentschev** [comment]
I think that would be good, but I definitely won't have the bandwidth for the next month or so.

**Comment by vyasr** [comment]
That's fine. I'm going to close this issue because the underlying ask (fixing UCX support) has been addressed. We now have working wheels for both ucxx and ucx-py, and the cuml testing provides us a good indication of that. The cugraph testing is a nice-to-have, but at this point I'd say it tells us more about whether cugraph's usage of UCX than about the correctness of our ucx* wheels. Feel free to add the cugraph testing whenever you have time.

**Comment by pentschev** [comment]
Thanks @vyasr and @jameslamb for all the work to make UCX wheels working (again?)! 😄

**Comment by jameslamb** [comment]
Thanks @pentschev for your patience and for all the help!
